runtime/runtime-params-estimator/estimator-warehouse/src/check.rs

```diff
@@ -42,41 +38,20 @@ pub(crate) enum Status {
     // Critical = 2,
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub(crate) enum Notice {
     RelativeChange(RelativeChange),
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub(crate) struct RelativeChange {
     pub estimation: String,
     pub before: f64,
     pub after: f64,
 }
 
 pub(crate) fn check(db: &Db, config: &CheckConfig) -> anyhow::Result<()> {
-    let (commit_a, commit_b) = match (&config.commit_a, &config.commit_b) {
-        (Some(a), Some(b)) => (a.clone(), b.clone()),
-        (None, None) => {
-            let mut commits = EstimationRow::commits_sorted_by_date(db, Some(config.metric))?;
-            if commits.len() < 2 {
-                anyhow::bail!("Need data for at least 2 commits to perform comparison.");
-            }
-            (commits.pop().unwrap().0, commits.pop().unwrap().0)
-        }
-        _ => anyhow::bail!("You have to either specify both commits for comparison or neither."),
-    };
-
-    let estimations = if config.estimations.len() > 0 {
-        config.estimations.clone()
-    } else {
-        let rows_a = EstimationRow::select_by_commit_and_metric(db, &commit_a, config.metric)?;
-        let rows_b = EstimationRow::select_by_commit_and_metric(db, &commit_b, config.metric)?;
-        let estimations_a = rows_a.into_iter().map(|row| row.name).collect::<BTreeSet<_>>();
-        let estimations_b = rows_b.into_iter().map(|row| row.name).collect::<BTreeSet<_>>();
-        estimations_a.intersection(&estimations_b).cloned().collect()
-    };
-    let warnings = estimation_changes(db, &estimations, &commit_a, &commit_b, 0.1, Metric::Time)?;
+    let (commit_before, commit_after, warnings) = inner_check(config, db)?;
     for warning in &warnings {
         println!("{warning:?}");
```

Comment:
Not a strong reason but it seems like a bit of an overkill to pull in dependencies etc. But if you can give me one reason why it would be better to use `tracing`, I am happy to use it.

(Note that in logging terminology, these are not warnings. This is just the output of the check command.)